gstd/src/prog/default_salt.rs

```diff
@@ -0,0 +1,36 @@
+// This file is part of Gear.
```

Comment:
First of all, I don't like the `default_salt` naming. It's pretty similar to `Default` trait for me.

Merge it into mod or provide mod like `salt_generator` or `generator` or even `salty`. Integrate it by default into `create_program` method, like `create_program_with_salt` or provide some beauty builder pattern solution if you don't like large function names.

Reply:
@montaglue originally there was an idea to create some manager, which creates new programs and has default salt management under the hood, generates salt privately. I suggest you going this way, because default salt management means that programmer doesn't think about salt at all, not that he should request default salt.

Reply:
I make ProgramGenerator that creates programs, I suppose new version of interface will be comfortable to use.